Model/Smartcalcs.php

```diff
@@ -543,21 +576,24 @@ private function _getLineItems(
      */
     private function _getNexusAddresses($storeId)
     {
-        $nexusAddresses = $this->nexusFactory->create()->getCollection()->addStoreFilter($storeId);
-        $addresses = [];
-
-        foreach ($nexusAddresses as $nexusAddress) {
-            $addresses[] = [
-                'id' => $nexusAddress->getId(),
-                'country' => $nexusAddress->getCountryId(),
-                'zip' => $nexusAddress->getPostcode(),
-                'state' => $nexusAddress->getRegionCode(),
-                'city' => $nexusAddress->getCity(),
-                'street' => $nexusAddress->getStreet()
-            ];
-        }
+        $nexusCollection = $this->nexusFactory->create()->getCollection()->addStoreFilter($storeId);
+        return array_map([$this, '_getNexusAddress'], $nexusCollection);
```

Comment:
I'm getting an error here:
[2022-01-31 17:31:28] main.CRITICAL: Report ID: webapi-61f81cf011158; Message: Warning: array_map(): Expected parameter 2 to be an array, object given in /var/www/html/app/code/Taxjar/SalesTax/Model/Smartcalcs.php on line 580 {"exception":"[object] (Exception(code: 0): Report ID: webapi-61f81cf011158; Message: Warning: array_map(): Expected parameter 2 to be an array, object given in /var/www/html/app/code/Taxjar/SalesTax/Model/Smartcalcs.php on line 580 at /var/www/html/vendor/magento/framework/Webapi/ErrorProcessor.php:208, Exception(code: 0): Warning: array_map(): Expected parameter 2 to be an array, object given in /var/www/html/app/code/Taxjar/SalesTax/Model/Smartcalcs.php on line 580 at /var/www/html/vendor/magento/framework/App/ErrorHandler.php:61)"} []

Reply:
🤦 I believe this was due to a bad "soft" reset.. I've abstracted these nexus-related operations into a separate Helper class and added test coverage in `\Taxjar\SalesTax\Test\Unit\Helper\NexusTest.php` to prevent regression.